pandas/indexes/numeric.py

```diff
@@ -22,6 +22,36 @@ class NumericIndex(Index):
     """
     _is_numeric_dtype = True
 
+    def __new__(cls, data=None, dtype=None, copy=False, name=None,
+                fastpath=False):
+
+        if fastpath:
+            return cls._simple_new(data, name=name)
+
+        if dtype and np.dtype(dtype) != cls._default_dtype:
```

Comment:
Are you sure we want to silently upcast e.g. from ``int32`` to ``int64`` _when the user explicitly asked_ (``dtype`` argument) an ``int32``?! I'd rather raise than fake.